From 7d7133d6b0eee7479e7cb6548cc9bee24891a26a Mon Sep 17 00:00:00 2001
From: PyScotch <pyscotch@noreply.invalid>
Date: Tue, 13 Oct 2026 16:07:11 +0000
Subject: [PATCH 07/11] Bugfix: reject unknown output format codes in `gcv` and
 `mcv` [report R. Perry]

The "-o" option loop scans the output format table but tested the
end of the input format table, so that an unknown output format code
was silently ignored and the default output format used instead.

Refs: CLI-14
---
 src/scotch/gcv.c | 8 ++++----
 src/scotch/mcv.c | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/scotch/gcv.c b/src/scotch/gcv.c
index 0618e44..60c0aad 100644
--- a/src/scotch/gcv.c
+++ b/src/scotch/gcv.c
@@ -1,4 +1,4 @@
-/* Copyright 2004,2007,2008,2010-2012,2014,2018,2019,2023,2024 Universite de Bordeaux, INRIA & CNRS
+/* Copyright 2004,2007,2008,2010-2012,2014,2018,2019,2023,2024,2026 Universite de Bordeaux, INRIA & CNRS
 **
 ** This file is part of the Scotch software package for static mapping,
 ** graph partitioning and sparse matrix ordering.
@@ -62,7 +62,7 @@
 /**                # Version 6.0  : from : 01 jan 2012     **/
 /**                                 to   : 24 sep 2019     **/
 /**                # Version 7.0  : from : 21 jan 2023     **/
-/**                                 to   : 10 sep 2024     **/
+/**                                 to   : 13 oct 2026     **/
 /**                                                        **/
 /************************************************************/
 
@@ -175,7 +175,7 @@ char *                      argv[])
           if (C_inpFormatTab[j].code == '\0')
             errorPrint ("main: unprocessed option '%s'", argv[i]);
           break;
-        case 'O' :                               /* Select input file type */
+        case 'O' :                               /* Select output file type */
         case 'o' :
           for (j = 0; C_outFormatTab[j].code != '\0'; j ++) { /* Find proper format code */
             if (C_outFormatTab[j].code == argv[i][2]) {
@@ -184,7 +184,7 @@ char *                      argv[])
               break;
             }
           }
-          if (C_inpFormatTab[j].code == '\0')
+          if (C_outFormatTab[j].code == '\0')
             errorPrint ("main: unprocessed option '%s'", argv[i]);
           break;
         case 'V' :
diff --git a/src/scotch/mcv.c b/src/scotch/mcv.c
index dda38e6..6716222 100644
--- a/src/scotch/mcv.c
+++ b/src/scotch/mcv.c
@@ -1,4 +1,4 @@
-/* Copyright 2004,2007,2008,2010-2012,2014,2018,2019,2023,2024 IPB, Universite de Bordeaux, INRIA & CNRS
+/* Copyright 2004,2007,2008,2010-2012,2014,2018,2019,2023,2024,2026 IPB, Universite de Bordeaux, INRIA & CNRS
 **
 ** This file is part of the Scotch software package for static mapping,
 ** graph partitioning and sparse matrix ordering.
@@ -47,7 +47,7 @@
 /**                # Version 6.0  : from : 01 jan 2012     **/
 /**                                 to   : 24 sep 2019     **/
 /**                # Version 7.0  : from : 21 jan 2023     **/
-/**                                 to   : 10 sep 2024     **/
+/**                                 to   : 13 oct 2026     **/
 /**                                                        **/
 /************************************************************/
 
@@ -148,7 +148,7 @@ char *                      argv[])
           if (C_inpFormatTab[j].code == '\0')
             errorPrint ("main: unprocessed option '%s'", argv[i]);
           break;
-        case 'O' :                               /* Select input file type */
+        case 'O' :                               /* Select output file type */
         case 'o' :
           for (j = 0; C_outFormatTab[j].code != '\0'; j ++) { /* Find proper format code */
             if (C_outFormatTab[j].code == argv[i][2]) {
@@ -157,7 +157,7 @@ char *                      argv[])
               break;
             }
           }
-          if (C_inpFormatTab[j].code == '\0')
+          if (C_outFormatTab[j].code == '\0')
             errorPrint ("main: unprocessed option '%s'", argv[i]);
           break;
         case 'V' :
-- 
2.47.3

